pallets/settlement/src/lib.rs

```diff
@@ -1522,55 +1524,101 @@ impl<T: Trait> Module<T> {
             instruction_id,
-            portfolios_set,
-            secondary_key.as_ref(),
-        )?;
+        )
+    }
 
-        // Execute the instruction if conditions are met
-        let affirms_pending = Self::instruction_affirms_pending(instruction_id);
-        let weight_for_instruction_execution = Self::is_instruction_executed(
-            affirms_pending,
+    /// It is used for general purpose settlement where instruction get affirmation &
+    /// may schedule for the next block execution or it is assumed that it is already scheduled.
+    pub fn affirm_and_may_schedule_instruction(
+        origin: <T as frame_system::Trait>::Origin,
+        instruction_id: u64,
+        portfolios: Vec<PortfolioId>,
+    ) -> DispatchResult {
+        Self::base_affirm_instruction(origin, instruction_id, portfolios)?;
+        // Schedule the instruction if conditions are met
+        Self::maybe_schedule_instruction(
+            Self::instruction_affirms_pending(instruction_id),
             Self::instruction_details(instruction_id).settlement_type,
             instruction_id,
-        );
+        )
+    }
+
+    /// It is a specialized function to affirm & execute the instruction when all affirmations
+    /// have received.
+    ///
+    /// NB - It is recommended to use this function only in the STO pallet to support the DVP settlement.
+    pub fn affirm_and_execute_instruction(
+        origin: <T as frame_system::Trait>::Origin,
+        instruction_id: u64,
+        portfolios: Vec<PortfolioId>,
+    ) -> DispatchResult {
+        Self::base_affirm_instruction(origin, instruction_id, portfolios)?;
+        Self::is_execute_instruction(
+            instruction_id,
+            Self::instruction_affirms_pending(instruction_id),
+            Self::instruction_details(instruction_id).settlement_type,
+        )
+    }
+
+    /// It is a specialized function to affirm with receipts & execute the instruction when all affirmations
+    /// have received.
+    ///
+    /// NB - It is recommended to use this function only in the STO pallet to support the DVP settlement.
+    pub fn affirm_with_receipts_and_execute_instruction(
+        origin: <T as frame_system::Trait>::Origin,
+        instruction_id: u64,
+        receipt_details: Vec<ReceiptDetails<T::AccountId, T::OffChainSignature>>,
+        portfolios: Vec<PortfolioId>,
+    ) -> DispatchResult {
+        Self::base_affirm_with_receipts(origin, instruction_id, receipt_details, portfolios)?;
+        Self::is_execute_instruction(
+            instruction_id,
+            Self::instruction_affirms_pending(instruction_id),
+            Self::instruction_details(instruction_id).settlement_type,
+        )
+    }
 
-        weight_for_instruction_execution.map(|info| {
-            info.actual_weight
-                .map(|w| w.saturating_add(weight_for::weight_for_affirmation_instruction::<T>()))
-                .into()
-        })
+    fn is_execute_instruction(
```

Comment:
I'd suggest renaming it to something else. Maybe `execute_settle_on_affirmation_instruction`.

`is` prefix does not really make sense for what the function does.